app/backend/approaches/retrievethenreadvision.py

```diff
@@ -81,24 +81,22 @@ async def run(
 
         overrides = context.get("overrides", {})
         auth_claims = context.get("auth_claims", {})
-        has_text = overrides.get("retrieval_mode") in ["text", "hybrid", None]
-        has_vector = overrides.get("retrieval_mode") in ["vectors", "hybrid", None]
-        vector_fields = overrides.get("vector_fields", ["embedding"])
-
-        include_gtpV_text = overrides.get("gpt4v_input") in ["textAndImages", "texts", None]
-        include_gtpV_images = overrides.get("gpt4v_input") in ["textAndImages", "images", None]
-
-        use_semantic_captions = True if overrides.get("semantic_captions") and has_text else False
+        use_text_search = overrides.get("retrieval_mode") in ["text", "hybrid", None]
+        use_vector_search = overrides.get("retrieval_mode") in ["vectors", "hybrid", None]
+        use_semantic_ranker = True if overrides.get("semantic_ranker") else False
+        use_semantic_captions = True if overrides.get("semantic_captions") else False
         top = overrides.get("top", 3)
         minimum_search_score = overrides.get("minimum_search_score", 0.0)
         minimum_reranker_score = overrides.get("minimum_reranker_score", 0.0)
         filter = self.build_filter(overrides, auth_claims)
-        use_semantic_ranker = overrides.get("semantic_ranker") and has_text
 
-        # If retrieval mode includes vectors, compute an embedding for the query
+        vector_fields = overrides.get("vector_fields", ["embedding"])
+        include_gtpV_text = overrides.get("gpt4v_input") in ["textAndImages", "texts", None]
```

Comment:
gptV? not gtpV?

Reply:
Existing typo, renamed for clarity across both files